src/request/derive-address/DeriveAddress.js

```diff
@@ -117,51 +56,49 @@ class DeriveAddress {
             if (e.message === 'Invalid key') {
                 TopLevelApi.setLoading(false);
                 this._passwordBox.onPasswordIncorrect();
-                return false;
+                return;
             }
             this._reject(new Errors.CoreError(e));
-            return false;
+            return;
         }
 
         if (!key) {
             this._reject(new Errors.KeyNotFoundError());
-            return false;
+            return;
         }
         const masterKey = /** @type {Nimiq.Entropy} */ (key.secret).toExtendedPrivateKey();
         const pathsToDerive = this._request.indicesToDerive.map(index => `${this._request.baseKeyPath}/${index}`);
 
-        this._identiconSelector.init(masterKey, pathsToDerive);
-        TopLevelApi.setLoading(false);
-        return true;
+        const derivedAddresses = this._generateAddresses(masterKey, pathsToDerive);
+        this._resolve(derivedAddresses);
     }
 
     async run() {
-        if (this._request.keyInfo.encrypted) {
-            window.location.hash = DeriveAddress.Pages.UNLOCK;
-        } else {
-            await this._onPasswordEntered();
-            window.location.hash = DeriveAddress.Pages.CHOOSE_IDENTICON;
-        }
+        window.location.hash = DeriveAddress.Pages.UNLOCK;
     }
 
     /**
-     * @param {string} address
+     * @param {Nimiq.ExtendedPrivateKey} masterKey
+     * @param {string[]} pathsToDerive
+     * @returns {KeyguardRequest.DerivedAddress[]}
      */
-    _openDetails(address) {
-        // eslint-disable-next-line no-new
-        new Identicon(address, this.$identicon);
-        // last space is necessary for the rendering to work properly with white-space: pre-wrap.
-        this.$address.textContent = `${address} `;
-
-        this.$chooseIdenticonPage.classList.add('account-details-open');
-    }
+    _generateAddresses(masterKey, pathsToDerive) {
+        /** @type { KeyguardRequest.DerivedAddress[] } */
+        const derivedAddresses = [];
+
+        for (let i = 0; i < pathsToDerive.length; i++) {
```

Comment:
could be 
```suggestion
        for (const path of pathsToDerive) {
```
and then replace the `pathsToDerive[i]` by `path`.

To me personally that reads better.